### ISSUE TITLE
Provider does not gracefully fail when no company files are available

### ISSUE THREAD

**Issue by anicholson** [davidlumley/omniauth-myob#4]
I realised that #3 occurred because the account associated with my MYOB keys has no company files associated. I don't think this is an error (albeit not very useful for me!), but as a developer I have no way of recovering so I can gracefully fail.

**Comment by anicholson** [comment]
Specifically, in my case:
- When the provider hits the base API endpoint, I get an empty JSON array back `[]` with status `200`. I take this to mean I have no company files associated with my account.
- The methods in the provider all assume that I must have at least one company file, and so reach into the `raw_info` expecting data to be there. When there isn't any, it fails **hard** (`NoMethodError` on `nil`) 

I should be able to recover from that and show an error message to my users, instead of a 500 with no visibility of what went wrong.

**Comment by davidlumley** [comment]
Should have time later this week to have a look, but feel free to submit a pull request in the mean time – thanks!

**Comment by davidlumley** [comment]
Finally had a chance to look into this.

Once you've authenticate against MYOB's API, you need a company file in order to access every other part of the API.

I think it makes sense to throw an exception that you're able to recover from gracefully. Thoughts?

**Comment by anicholson** [comment]
Yes, I think that's the best solution. Maybe a name like `NoCompanyFilesAvailableException` ?
